File: server/app/models/embedding_config.py
from sqlalchemy import Column, Integer, String, DateTime, Text
from datetime import datetime
from app.database import Base
from .base import LOCALTIME


class EmbeddingConfig(Base):
    __tablename__ = "embedding_configs"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    model_id = Column(String(100), nullable=False)
    base_url = Column(String(255), nullable=False)
    api_key = Column(String(255), nullable=False)
    description = Column(Text, nullable=False, default='')
    created_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, nullable=False)
    updated_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, onupdate=datetime.now, nullable=False)

File: server/internal/model/embedding_config.go
package model

import "time"

// EmbeddingConfig stores the configuration for the embedding API provider.
// Used for RAG (Retrieval-Augmented Generation) vector operations.
type EmbeddingConfig struct {
	ID          int64     `gorm:"primaryKey;autoIncrement;type:INTEGER PRIMARY KEY AUTOINCREMENT" json:"id"`
	Name        string    `gorm:"type:varchar(100);not null" json:"name"`
	ModelID     string    `gorm:"type:varchar(100);not null;column:model_id" json:"model_id"`
	BaseURL     string    `gorm:"type:varchar(255);not null;column:base_url" json:"base_url"`
	APIKey      string    `gorm:"type:varchar(255);not null;column:api_key" json:"api_key"`
	Description string    `gorm:"type:text;not null;default:''" json:"description"`
	CreatedAt   time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
	UpdatedAt   time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (EmbeddingConfig) TableName() string { return "embedding_configs" }
